test/common/modules/live_objects_helper.js

```diff
@@ -0,0 +1,172 @@
+     */
+    async initForChannel(helper, channelName) {
+      const rest = helper.AblyRest({ useBinaryProtocol: false });
+
+      const emptyCounter = await this._createAndSetOnMap(rest, channelName, {
+        mapObjectId: 'root',
+        key: 'emptyCounter',
+        createOp: this._counterCreateOp(),
+      });
+      const initialValueCounter = await this._createAndSetOnMap(rest, channelName, {
+        mapObjectId: 'root',
+        key: 'initialValueCounter',
+        createOp: this._counterCreateOp({ count: 10 }),
+      });
+      const referencedCounter = await this._createAndSetOnMap(rest, channelName, {
+        mapObjectId: 'root',
+        key: 'referencedCounter',
+        createOp: this._counterCreateOp({ count: 20 }),
+      });
+
+      const emptyMap = await this._createAndSetOnMap(rest, channelName, {
+        mapObjectId: 'root',
+        key: 'emptyMap',
+        createOp: this._mapCreateOp(),
+      });
+      const referencedMap = await this._createAndSetOnMap(rest, channelName, {
+        mapObjectId: 'root',
+        key: 'referencedMap',
+        createOp: this._mapCreateOp({ entries: { counterKey: { data: { objectId: referencedCounter.objectId } } } }),
+      });
+      const valuesMap = await this._createAndSetOnMap(rest, channelName, {
+        mapObjectId: 'root',
+        key: 'valuesMap',
+        createOp: this._mapCreateOp({
+          entries: {
+            stringKey: { data: { value: 'stringValue' } },
+            emptyStringKey: { data: { value: '' } },
+            bytesKey: {
+              data: { value: 'eyJwcm9kdWN0SWQiOiAiMDAxIiwgInByb2R1Y3ROYW1lIjogImNhciJ9', encoding: 'base64' },
+            },
+            emptyBytesKey: { data: { value: '', encoding: 'base64' } },
+            numberKey: { data: { value: 1 } },
+            zeroKey: { data: { value: 0 } },
+            trueKey: { data: { value: true } },
+            falseKey: { data: { value: false } },
+            mapKey: { data: { objectId: referencedMap.objectId } },
+          },
+        }),
+      });
+    }
+
+    async _createAndSetOnMap(rest, channelName, opts) {
+      const { mapObjectId, key, createOp } = opts;
+
+      const createResult = await this._stateRequest(rest, channelName, createOp);
+      await this._stateRequest(
+        rest,
+        channelName,
+        this._mapSetOp({ objectId: mapObjectId, key, data: { objectId: createResult.objectId } }),
+      );
+
+      return createResult;
+    }
+
+    _mapCreateOp(opts) {
+      const { objectId, entries } = opts ?? {};
+      const op = {
+        operation: {
+          action: ACTIONS.MAP_CREATE,
+          nonce: nonce(),
+          objectId,
+        },
+      };
+
+      if (entries) {
+        op.operation.map = { entries };
+      }
+
+      return op;
+    }
+
+    _mapSetOp(opts) {
+      const { objectId, key, data } = opts ?? {};
+      const op = {
+        operation: {
+          action: ACTIONS.MAP_SET,
+          nonce: nonce(),
```

Comment:
The nonce is not needed here, only `CREATE` ops require a nonce, other ops have to explicitly reference the `objectId`

Reply:
true, removed nonce for MAP_SET